engine/src/cubos/engine/plugins/tools/world_inspector.cpp

```diff
@@ -0,0 +1,42 @@
+#include <cubos/engine/plugins/tools/world_inspector.hpp>
+#include <imgui.h>
+
+#include <cubos/engine/plugins/imgui.hpp>
+#include <cubos/engine/plugins/tools/entity_selector.hpp>
+
+using namespace cubos;
+using namespace cubos::engine::plugins::tools;
+using namespace cubos::core::ecs;
+
+static void inspectWorld(World& world, EntitySelector& selector)
+{
+    ImGui::Begin("World Inspector");
+    if (!ImGui::IsWindowCollapsed())
+    {
+        int n = 0;
+        for (World::Iterator i = world.begin(); i != world.end(); ++i)
```

Comment:
I think you can the use the for-each syntax: `for (auto entity : world)`, not sure.

Reply:
Oh yeah, I always forget there's foreach in c++